src/os/lsof_utils.rs

```diff
@@ -16,46 +17,114 @@ pub struct RawConnection {
 
 lazy_static! {
     static ref CONNECTION_REGEX: Regex =
-        Regex::new(r"([^\s]+).*(TCP|UDP).*:(.*)->(.*):(\d*)(\s|$)").unwrap();
+        Regex::new(r"\[?([^\s\]]*)\]?:(\d+)->\[?([^\s\]]*)\]?:(\d+)").unwrap();
+    static ref LISTEN_REGEX: Regex = Regex::new(r"(.*):(.*)").unwrap();
 }
 
+fn parse_ip_addr(ip: &str) -> IpAddr {
+    if let Ok(v4addr) = ip.parse() {
+        IpAddr::V4(v4addr)
+    } else if let Ok(v6addr) = ip.parse() {
+        IpAddr::V6(v6addr)
+    } else {
+        panic!("{} is not a valid IP address", ip)
+    }
+}
+
+fn get_null_addr(ip_type: &str) -> &str {
+    if ip_type.contains('4') {
+        "0.0.0.0"
```

Comment:
We could maybe leverage the [`UNSPECIFIED`](https://doc.rust-lang.org/std/net/struct.Ipv4Addr.html#associatedconstant.UNSPECIFIED) constants, for IpV4 and IpV6.

Reply:
hmmm but we will have to do `UNSPECIFIED.to_string()` , as parsing of IP address happens later, we can't directly return an `IpAddr` object.

Reply:
I think we could add a `lazy_static` as `String` and take the reference here, but don't give this a lot of effort if it becomes cumbersome!

Reply:
🥺 Too much trouble... Maybe I can do it in future PRs